addons/account/models/account_reconcile_model.py

```diff
@@ -797,6 +797,7 @@ def _get_invoice_matching_rule_result(self, st_line, candidates, aml_ids_to_excl
             # A write-off must be applied if there are some 'new' lines to propose.
             if not lines_vals_list or any(not line_vals.get('id') for line_vals in lines_vals_list):
                 rslt['status'] = 'write_off'
+                rslt['write_off_vals'] = list(filter(lambda x: 'id' not in x, lines_vals_list))
```

Comment:
I don't really like iterating twice the same list `any(not line_vals.get('id') for line_vals in lines_vals_list)` & `list(filter(lambda x: 'id' not in x, lines_vals_list))`. Don't you think it would be cleaner with a variable?